File: 2017-2018 winter/теория чисел/A.py
def gcd(a, b):
    if b == 0:
        return (a, 1, 0)
    else:
        d, x1, y1 = gcd(b, a % b)
        return (d, y1, x1 - y1 * (a // b))
    
a, b = list(map(int, input().split()))
a %= b
d, x, y = gcd(a, b)
if d != 1:
    print(-1)
else:
    print(x % b)

File: 2017-2018 winter/теория чисел/A.cpp
#define _CRT_SECURE_NO_WARNINGS
#include<iostream>
#include<fstream>
#include<vector>
#include<algorithm>
#include<string>
#include<stack>
#include<cmath>
#include<map>
#include<set>
#include<climits>
#include<queue>

using namespace std;

int gcd(int a, int b)
{
	while (b > 0)
	{
		int tmp = a;
		a = b;
		b = tmp % b;
	}
	return a;
}

//FILE* in = fopen("inv.in", "r");
//FILE* out = fopen("inv.out", "w");

FILE* in = fopen("input.txt", "r");
FILE* out = fopen("output.txt", "w");

int main()
{
	long long n, m;
	fscanf(in, "%lld %lld", &n, &m);
	return 0;
}